### PR TITLE
feat: documentation generation for edgex-global-pipelines

### PR DESCRIPTION
**Pull request by cjoyv**
feat: documentation generation for edgex-global-pipelines

Signed-off-by: Chinu Joy <chinu.joy@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Documentation content changes
- [ ] Other... Please describe:

## Sandbox Testing
Test Links : https://jenkins.edgexfoundry.org/sandbox/view/All/job/Functional-Testing/job/github-pages-publish/

### CODE REVIEW

**Review by soda480 — DISMISSED** [review]
LGTM

File: Jenkinsfile [hunk 1]
```diff
@@ -93,6 +93,71 @@ pipeline {
             }
         }
 
+        stage('Generate Documentation') {
+            agent {
+                docker {
+                    image 'gradle'
```
Comment (ernestojeda) [inline Jenkinsfile:99]: Can we pin this to a specific version please.

File: build.gradle [hunk 2]
```diff
@@ -31,9 +31,19 @@ repositories {
...
+    groovy {
+      srcDirs = ['vars/']
+    }
+  }
+}
+
 dependencies {
   def withoutIcu = { exclude group: 'com.ibm.icu', module: 'icu4j' }
 
+  compile "org.jenkins-ci.plugins.workflow:workflow-cps-global-lib:2.15@jar"
```
Comment (ernestojeda) [inline build.gradle:45]: It's interesting this needed to be added. I will take a look at this at a later time

File: vars/edgeXDocGenerator.groovy [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import groovy.io.FileType
```
Comment (ernestojeda) [inline vars/edgeXDocGenerator.groovy:1]: Can this file be moved out of the `vars` directory? Into a directory like `src/main/groovy` or similar. Only global library functions should be in the `vars` directory

**Review by ernestojeda — CHANGES_REQUESTED** [review]
See my comments

**Review by ernestojeda — APPROVED** [review]
👍  LGTM. This is a great addition to the project.

**Pull request merged**